### PR TITLE
Create request endpoint

### LINKED ISSUE #46: [BACKEND] Implement API to create request.

### PR DESCRIPTION
**Pull request by yililiu8**

### CODE REVIEW

File: backend/api/request_create.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			}
+			return
+		}
+
+		arg = db.CreateRequestParams{
+			UserID:      authPayload.UserID,
+			CommunityID: sql.NullInt64{Int64: req.CommunityID, Valid: true},
+			StoreID:     sql.NullInt64{Int64: *req.StoreID, Valid: true},
+		}
+	}
```
Comment (yililiu8) [inline backend/api/request_create.go]: Let me know if there's a better way to deal with optional arguments

Comment (git-adithyanair) [reply backend/api/request_create.go]: Nah you're good, that's how its done

File: backend/api/server.go [hunk 2]
```diff
@@ -56,6 +56,9 @@ func (server *Server) setupRouter() {
 	protectedRoutes.POST("/community", server.CreateCommunity)
 	protectedRoutes.GET("/community/:id", server.GetCommunity)
 
+	// Request routes.
+	protectedRoutes.POST("/request/create", server.CreateRequest)
```
Comment (git-adithyanair) [inline backend/api/server.go]: Make the endpoint just `/request` to match the other create requests.

File: backend/api/request_create.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			RequestID:      request.ID,
+			QuantityType:   item.QuantityType,
+			Quantity:       item.Quantity,
+			PreferredBrand: sql.NullString{String: item.PreferredBrand, Valid: len(item.PreferredBrand) > 0},
+			Image:          sql.NullString{String: item.Image, Valid: len(item.Image) > 0},
+			ExtraNotes:     sql.NullString{String: item.ExtraNotes, Valid: len(item.ExtraNotes) > 0},
+		}
+
+		_, err := server.queries.CreateItem(ctx, arg)
+		if err != nil {
```
Comment (git-adithyanair) [inline backend/api/request_create.go]: Let's ignore errors here, we don't want to return from the handler while items are being added.

File: backend/api/request_create.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			if err == sql.ErrNoRows {
+				err := errors.New("No store exists with given ID.")
+				ctx.JSON(http.StatusNotFound, errorResponse(err))
+			} else {
+				ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+			}
+			return
+		}
+
+		arg = db.CreateRequestParams{
```
Comment (git-adithyanair) [inline backend/api/request_create.go]: Is it possible to just do `arg.StoreID = sql.NullInt64{Int64: *req.StoreID, Valid: true}` here?

File: backend/api/request_create.go (hunk 3 above)
Comment (yililiu8) [reply backend/api/request_create.go]: Ah yes good catch

File: backend/api/request_create.go (hunk 4 above)
Comment (yililiu8) [reply backend/api/request_create.go]: oh yes did not know you could do that but it seems to work

**Pull request merged**